cmds/exp/traceroute/traceroute_test.go

```diff
@@ -0,0 +1,148 @@
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/traceroute"
+)
+
+func TestParseFlags(t *testing.T) {
+	for _, tt := range []struct {
+		name    string
+		cmdline []string
+		exp     *traceroute.Flags
+		err     error
+	}{
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-4", "-m", "icmp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp4",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-4", "--icmp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp4",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-4", "-m", "tcp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp4",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-4", "--tcp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp4",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-4", "-m", "udp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp4",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-4", "--udp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp4",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-6", "-m", "icmp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp6",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-6", "--icmp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp6",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-6", "-m", "tcp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp6",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-6", "--tcp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp6",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-6", "-m", "udp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp6",
+			},
+		},
+		{
+			name:    "Simple",
+			cmdline: []string{"progName", "-6", "--udp", "www.google.com"},
+			exp: &traceroute.Flags{
+				Host:   "www.google.com",
+				Module: "icmp",
+				Proto:  "icmp6",
+			},
+		},
+		{
+			name:    "Simple",
```

Comment:
all test cases have the same name, it's better to rename or remove

Reply:
Renamed